### DISCUSSION TITLE
Can atomColourPalette be ignored by some other logic?

### DISCUSSION THREAD

**Discussion by heberleh** [rdkit/rdkit#4871] [Q&A] [answered]
I am trying to set all atoms in black following #4853, but some won't change.

In most cases, `Br` and `Cl` are colored in brown and green. `N` and `O` seem to work, always in black.

To test it, I set the labels to be the same index of the color palette.

Cl-14 and Cl-12:
`N#Cc1c[nH]cc1c1cccc(Cl)c1Cl`
<img width="250" alt="Screenshot 2022-01-12 at 11 51 57" src="https://user-images.githubusercontent.com/4249107/149127723-738b309c-6236-4804-a8da-fe9d6738bf67.png">

Here is the atomColourPalette:
<img width="200" alt="Screenshot 2022-01-12 at 11 51 57" src="https://user-images.githubusercontent.com/4249107/149128107-5cf923f1-e4ac-4807-ad7d-aee8a4240676.png">

Another example, atom 3 (Br):
`Cc1c(Br)c2ccc(F)cc2n1S(=O)(=O)c1ncn(S(=O)(=O)N(C)C)n1`
<img width="350" alt="Screenshot 2022-01-12 at 11 51 57" src="https://user-images.githubusercontent.com/4249107/149128477-ce7bdd3f-518c-403b-bfd4-c07cbd0c5111.png">

Here it works:
`Cn1cc(C(=O)Nc2cccc3c2C2CCC3C2=C(Cl)Cl)c(C(F)F)n1`
<img width="350" alt="Screenshot 2022-01-12 at 11 51 57" src="https://user-images.githubusercontent.com/4249107/149131004-d8c9e2bf-aeeb-47cc-9620-a24682d9741c.png">

You can reproduce using this code:

```javascript
  var smiles = "N#Cc1c[nH]cc1c1cccc(Cl)c1Cl";
  var mol = RDKitModule.get_mol(smiles);
  var mdetails = {};
  mdetails['atoms']=[0,1,10];
  mdetails['explicitMethyl'] = true;
  mdetails['addAtomIndices'] = true;
  mdetails['legend']='aspirin';
  mdetails['atomColourPalette']={
            "0": [0, 0, 0],
            "1": [0, 0, 0],
            "2": [0, 0, 0],
            "3": [0, 0, 0],
            "4": [0, 0, 0],
            "5": [0, 0, 0],
            "6": [0, 0, 0],
            "7": [0, 0, 0],
            "8": [0, 0, 0],
            "9": [0, 0, 0],
            "10": [0, 0, 0],
            "11": [0, 0, 0],
            "12": [0, 0, 0],
            "13": [0, 0, 0],
            "14": [0, 0, 0],
        };
  var canvas = document.getElementById("canvas-4");
  mol.draw_to_canvas_with_highlights(canvas, JSON.stringify(mdetails));
```

You can quickly run it here, in the 4th input box. Only `N`s become black.
https://michelml.github.io/rdkit-js/examples/vanilla-javascript/#introduction

**Comment by heberleh** [comment, marked as answer]
Please close/delete this... It was a misunderstanding.
Because the Hash uses Integer, I thought they refer to atom-index (0, 1, 2, ...) as in the atom-graph (which would make it more flexible, and there could be a function to calculate this vector based on atom-names), and not to atom-name (C, N, ...).

**Reply by ptosco** [reply]
@heberleh Correct, as you have already found out the number is the atomic number, not the atom index. This is why in my original example I generated an `atomColourPalette` covering all elements in the periodic table to make sure that any possible would be painted in black.
